includes/alert.js

```diff
@@ -0,0 +1,410 @@
+    // There's nothing fancy about it, and much of it will still work
+    // correctly without JavaScript enabled. However, to offer a slicker
+    // visual user experience, we use BackboneJS to progressively enhance
+    // the static HTML content.
+    //
+    // Backbone helps us structure our interface code into "Views" that
+    // can be composed together. This means one View can contain another
+    // View inside it. Each interface element that we display in the HTML
+    // (like buttons, forms, and so on) are associated with a View at
+    // runtime by code in this file. Interactions that require JavaScript
+    // to work, like showing and hiding dialogues or other visual effects
+    // of a similar nature, are performed by calling methods on the View
+    // objects defined below.
+
+    // --------------------- //
+    // VIEWS (UI components) //
+    // --------------------- //
+
+    // Gotta declare 'em all.
+    var ActivateAlertForm,
+        ChooseTeamsPanel,
+        ContextualAlertButton,
+        ContextualAlertModal,
+        ImmediateAlertButton,
+        SubmittingAlertModal,
+        TimedAlertButton,
+        TimedAlertModal;
+
+    /**
+     * Activate Alert Form
+     *
+     * This is the containing element for the "Activate Alert" screen.
+     *
+     * @see {@link https://github.com/betterangels/buoy/wiki/Alerts#accessing-the-activate-alert-screen}
+     */
+    BUOY.Views.ActivateAlertForm = Backbone.View.extend({
+        'events': {
+            'submit': 'handleSubmit'
+        }
+    });
+
+    /**
+     * Immediate Alert Button
+     *
+     * @see {@link https://github.com/betterangels/buoy/wiki/Alerts#immediate-alerts}
+     */
+    BUOY.Views.ImmediateAlertButton = Backbone.View.extend({
+        'events': {
+            'click': function () {
+                this.$el.prop('disabled', true); // prevent double-clicks
+            }
+        }
+    });
+
+    /**
+     * Contextual Alert Button
+     *
+     * @see {@link https://github.com/betterangels/buoy/wiki/Alerts#contextual-alerts}
+     */
+    BUOY.Views.ContextualAlertButton = Backbone.View.extend({
+        'events': {
+            'click': function () {
+                ChooseTeamsPanel.show();
+                ContextualAlertModal.show();
+            }
+        }
+    });
+
+    /**
+     * Timed Alert Button
+     *
+     * @see {@link https://github.com/betterangels/buoy/wiki/Alerts#timed-alerts}
+     */
+    BUOY.Views.TimedAlertButton = Backbone.View.extend({
+        'events': {
+            'click': function () {
+                ChooseTeamsPanel.show();
+                TimedAlertModal.show();
+            }
+        }
+    });
+
+    /**
+     * Contextual Alert Modal
+     *
+     * This is a Bootstrap Modal.
+     *
+     * @see {@link https://getbootstrap.com/javascript/#modals}
+     */
+    BUOY.Views.ContextualAlertModal = Backbone.View.extend({
+        'events': {
+            'click button.btn-success': function (e) {
```

Comment:
Yeah, this is [conventional Backbone-style event delegation](http://backbonejs.org/#View-delegateEvents). So, you'll just have to suck up, I'm afraid. One thing we can do (and that we *are* doing) to make this a little less gross is that we're using conventional markup and behaviors provided by Bootstrap (as linked in the modal View's JSDoc docblock) to make sure that this Backbone component *always* knows about its own markup. The point here being that the markup itself will not change, because it's defined by Bootstrap, and so the Backbone View can know about the underlying markup of the interface component it's responsible, without relying on anything else in the code.

After trying it out (in 032bc1f53bf10f55c47b874dc6317fd7a27a2830, c86f8cda23cc8bb3e1163ec456b576993adf4589, and 533d82aca7f3fbd5daa5567475ae4313ed9942b3), I have a lot *more* distaste for the `buoy_dom_hooks` pattern thingy because it does not actually DRY the code and, moreover, it introduces a PHP dependency into the JavaScript. That is far, far worse, IMHO, than simply maintaining a conventional mapping between a Backbone View component and the HTML DOM for which it is responsible, which feels much more in keeping with [Single Responsibility](https://en.wikipedia.org/wiki/Single_responsibility_principle) anyway. It isn't perfect but dramatically reduces our edit-per-change ratio (how many places in the code we need to edit to maintain equivalent functionality for an arbitrary single change), which is paramount to this refactor in the first place.